### PR TITLE
ORM/HR/Panache: support `with` and `where` queries

### LINKED ISSUE #36496: Panache simplfied query + CTE with
### Describe the bug

Hibernate ORM panache does not recognize syntax starting with **"WITH"** according to the syntax provided in Hibernate 2.0 for enabling recursive queries (CTE). 

In fact, the query "**WITH** xxx AS (Select ...) Select * FROM xxx"  is replaced by "FROM *** WHERE **WITH**..." which is incorrect.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by FroMage**
Fixes #36496

### CODE REVIEW

File: docs/src/main/asciidoc/hibernate-orm-panache.adoc
Comment (loicmathieu) [inline docs/src/main/asciidoc/hibernate-orm-panache.adoc:1]: We have the same comment in the JavaDoc of the package-info.java, please made the same modification here: https://github.com/quarkusio/quarkus/blob/main/extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/package-info.java#L60

File: docs/src/main/asciidoc/hibernate-reactive-panache.adoc
Comment (loicmathieu) [inline docs/src/main/asciidoc/hibernate-reactive-panache.adoc:1]: We have the same comment in the JavaDoc of the package-info.java, please made the same modification here: https://github.com/quarkusio/quarkus/blob/main/extensions/panache/hibernate-reactive-panache/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/package-info.java#L56

**Review by loicmathieu — CHANGES_REQUESTED** [review]
You added test for a query that select with `WITH` but not there, can you also add one?

File: docs/src/main/asciidoc/hibernate-orm-panache.adoc
Comment (FroMage) [reply docs/src/main/asciidoc/hibernate-orm-panache.adoc:1]: Well spotted indeed!

**Comment by FroMage** [comment]
> You added test for a query that select with WITH but not there, can you also add one?

Not sure what you mean by `not there`?

**Comment by loicmathieu** [comment]
> Not sure what you mean by not there?

I mean not WHERE

**Comment by FroMage** [comment]
I did add tests for `where`, though.

File: integration-tests/hibernate-orm-panache/src/main/java/io/quarkus/it/panache/TestEndpoint.java [hunk 1]
```diff
@@ -85,6 +85,9 @@ public String testModel() {
         Assertions.assertEquals(1, Person.count("name = :name", Parameters.with("name", "stef").map()));
         Assertions.assertEquals(1, Person.count("name = :name", Parameters.with("name", "stef")));
         Assertions.assertEquals(1, Person.count("name", "stef"));
+        Assertions.assertEquals(1, Person.count("from Person2 where name = ?1", "stef"));
+        Assertions.assertEquals(1, Person.count("where name = ?1", "stef"));
```
Comment (FroMage) [inline integration-tests/hibernate-orm-panache/src/main/java/io/quarkus/it/panache/TestEndpoint.java:89]: Here

File: integration-tests/hibernate-orm-panache/src/main/java/io/quarkus/it/panache/TestEndpoint.java [hunk 2]
```diff
@@ -110,6 +113,10 @@ public String testModel() {
         Assertions.assertEquals(1, persons.size());
         Assertions.assertEquals(person, persons.get(0));
 
+        persons = Person.find("where name = ?1", "stef").list();
```
Comment (FroMage) [inline integration-tests/hibernate-orm-panache/src/main/java/io/quarkus/it/panache/TestEndpoint.java:116]: And here

File: integration-tests/hibernate-orm-panache/src/main/java/io/quarkus/it/panache/TestEndpoint.java (hunk 1 above)
Comment (loicmathieu) [reply integration-tests/hibernate-orm-panache/src/main/java/io/quarkus/it/panache/TestEndpoint.java:89]: Great! I read the PR too quickly ;)

File: integration-tests/hibernate-orm-panache/src/main/java/io/quarkus/it/panache/TestEndpoint.java (hunk 2 above)
Comment (loicmathieu) [reply integration-tests/hibernate-orm-panache/src/main/java/io/quarkus/it/panache/TestEndpoint.java:116]: Great! I read the PR too quickly ;)

**Review by loicmathieu — APPROVED** [review]
LGTM

**Pull request merged**